developer/src/kmc-kmn/src/compiler/compiler.ts

```diff
@@ -47,35 +47,72 @@ const baseOptions: CompilerOptions = {
  */
 let callbackProcIdentifier = 0;
 
+/**
+ * The wrapped functions
+ */
+class WrappedWasmFuncs {
+  compileKeyboardFile?: (a0: string, a1: string, a2: number, a3: number, a4: number, a5: string) => boolean;
+  parseUnicodeSet?: (a0: string, a1: number, a2: number) => number;
+  setCompilerOptions?: (shouldAddCompilerVersion: number) => boolean;
+
+  constructor(wasmModule: any) {
+    this.compileKeyboardFile = wasmModule.cwrap('kmcmp_Wasm_CompileKeyboardFile', 'boolean', ['string', 'string', 'number', 'number', 'number', 'string']);
+    this.parseUnicodeSet = wasmModule.cwrap('kmcmp_Wasm_ParseUnicodeSet', 'number', ['string', 'number', 'number']);
+    this.setCompilerOptions = wasmModule.cwrap('kmcmp_Wasm_SetCompilerOptions', 'boolean', ['number']);
+  }
+
+  /**
+   * @returns true if the functions are setup ok
+   */
+  get ok(): boolean {
+    return this.parseUnicodeSet !== undefined
+      && this.setCompilerOptions !== undefined
+      && this.compileKeyboardFile !== undefined;
+  }
+};
+
 export class Compiler {
   wasmModule: any;
-  compileKeyboardFile: any;
-  setCompilerOptions: any;
   callbackName: string;
   callbacks: CompilerCallbacks;
+  wasm: WrappedWasmFuncs;
 
   constructor() {
     this.callbackName = 'kmnCompilerCallback' + callbackProcIdentifier;
     callbackProcIdentifier++;
   }
 
-  public async init(): Promise<boolean> {
+  public async init(callbacks: CompilerCallbacks): Promise<boolean> {
+    if(!this.callbacks) {
+      this.callbacks = callbacks;
+    }
     if(!this.wasmModule) {
       this.wasmModule = await loadWasmHost();
```

Comment:
My suggestion is:

First, move `wasmModule` into `WrappedWasmFuncs` (probably rename that class then to say `WasmWrapper`).

Then, something like this:

```
try {
  this.wasm = new WasmWrapper();
} catch(e: Error) {
  this.callbacks.reportMessage(CompilerMessages.Fatal_MissingWasmModule({e}));
  return false;
}
```

Means we can just do `if(!this.wasm) { throw }` as the input condition for other functions in the class. I'm happy with a throw for those cases, because those are bugs in our code rather than an input or environment issue (e.g. missing wasm module is an environment issue and that's handled by `init()` which returns a polite error).